proto/mls/api/v1/mls.proto

```diff
@@ -190,3 +222,51 @@ message GetIdentityUpdatesResponse {
   // of the request
   repeated WalletUpdates updates = 1;
 }
+
+// Sort direction for queries
+enum SortDirection {
+  SORT_DIRECTION_UNSPECIFIED = 0;
+  SORT_DIRECTION_ASCENDING = 1;
+  SORT_DIRECTION_DESCENDING = 2;
+}
+
+// Pagination config for queries
+message PagingInfo {
+  SortDirection direction = 1;
+  uint32 limit = 2;
+  uint64 cursor = 3;
+}
+
+// Request for group message queries
+message QueryGroupMessagesRequest {
+  bytes group_id = 1;
+  PagingInfo paging_info = 2;
+}
+
+// Response for group message queries
+message QueryGroupMessagesResponse {
+  repeated xmtp.mls.message_contents.GroupMessage messages = 1;
+  PagingInfo paging_info = 2;
+}
+
+// Request for welcome message queries
+message QueryWelcomeMessagesRequest {
+  bytes installation_id = 1;
+  PagingInfo paging_info = 2;
+}
+
+// Response for welcome message queries
+message QueryWelcomeMessagesResponse {
+  repeated xmtp.mls.message_contents.WelcomeMessage messages = 1;
+  PagingInfo paging_info = 2;
+}
+
+// Request for following group messages
+message SubscribeGroupMessagesRequest {
+  bytes group_id = 1;
```

Comment:
This needs to allow for subscribing to multiple groups so that we can implement some equivalent of `StreamAllMessages`